disorder-jack/src/Disorder/Jack/Combinators.hs

```diff
@@ -232,13 +233,11 @@ frequency = \case
 
 -- | Randomly selects one of the values in the list.
 --   /The input list must be non-empty./
-elements :: [a] -> Jack a
-elements = \case
-  [] ->
-    Savage.error "Disorder.Jack.Combinators.elements: used with empty list"
-  xs -> do
-    n <- choose (0, List.length xs - 1)
-    pure $ xs List.!! n
+elements :: Foldable b => b a -> Jack a
+elements xs = do
+  when (length xs Savage.== 0) $ Savage.error "Disorder.Jack.Combinators.elements: used with empty foldable"
+  n <- choose (0, length xs - 1)
+  pure $ (toList xs) List.!! n
```

Comment:
It probably doesn't matter too much, but I'd be inclined to only walk the foldable once, something like:
```hs
elements :: Foldable a => a -> Jack a
elements xs =
  case toList xs of
    [] ->
      Savage.error "Disorder.Jack.Combinators.elements: used with empty foldable"
    ys -> do
      n <- choose (0, List.length ys - 1)
      pure $ ys List.!! n
```